### DISCUSSION TITLE
Help to improve myMPD

### DISCUSSION THREAD

**Discussion by jcorporation** [jcorporation/myMPD#167] [Announcements]
myMPD is a small hobby project of mine. I only work on it in my spare time, so any help is welcome.

- Please star this project to make it more popular
- Consider [donate a coffee](https://jcorporation.github.io/donate) to me.

## Test
Unfortunately I can not test all possible configurations and end devices. It is a great help for me if you test myMPD and report bugs and inconsistencies.

## Translate
Do you want myMPD in your language? Then help with the translation, details can be found <a href="https://jcorporation.github.io/myMPD/references/translating">here</a>.

You can also join the translation project at [POEditor](https://poeditor.com/join/project/Z54inZwdul).

## Write or proof read documentation
myMPD should have a good and error free [documentation](https://jcorporation.github.io/myMPD/).

## Provide feedback
Please consider to provide feedback, positive and negative for sure.

### Vote for enhancements
Vote for an enhancement, discuss the details or propose a new enhancement. Vote for a feature to push the probability for timely implementation up. Please open a new discussion as "Idea" for each proposed enhancement.

---
Many thanks and enjoy myMPD

**Comment by onkobu** [comment]
5.6 is a big improvement, installation works right away, even with different MYMPD_INSTALL_DIR – I preferrably move self compiled packages to /opt, and in conjunction with Apache as reverse proxy. From a developer's point of view moving towards quality checked code is the right direction.

**Comment by p3732** [comment]
Feedback, here it goes!

This is by far the best webview for MPD that I could find, I really like it and it is a huge improvement over ympd. I deployed this on a Raspi in a students café for the people behind the bar to select music, so if that is okay with you, I would like to forward the feedback that I got so far. Here is a short list:

- The enumerated pages at the bottom felt outdated to some, but I saw that there already are plans for it to be replaced in #200 
- The "add all" button did not clearly communicate what it does by its symbol alone, maybe expanding it to a label with "Add All" or similar would help here, given enough screen space is available.
- In the cover grid there seems to be no definite way to add a selected album to the queue. A button with "Add All" in the popover would be my suggestion here too.
- Due to the nature of the music collection I am working with, the cover grid contains a lot of albums with only a single song. This people found annoying. Maybe being able to define a minimum amount of songs for an album to appear in the cover grid could solve this.

I can follow up with more feedback as I receive it. If I should open up separate issues with more specific information on any of these items, I will gladly do so.

Cheers!

**Comment by jcorporation** [comment]
Thank you very much for your feedback! More feedback is always welcome.

> "Add all" buttons

I added a title to the buttons, they should popup on hover. I don't want text in the toolbars.

> Covergrid

You can already add the complete album to the queue or a playlist. If you click on the bottom of the album card and not on the image a popup opens with album actions.

> Defining a minimum amount of songs for covergrid

This is a good possibility, I add it to the ui enhancements issue

**Comment by Emilot** [comment]
Hi JC

Awesome new features and the UI is getting better version by version...

Can you add a field at "Stats" page with the IP address that myMPD use?

If someone browse from a PC/MAC can login with .local address but as you know from an Android device he/she can't. Only with IP address.

Thank you for your work again.

**Comment by jcorporation** [comment]
> Can you add a field at "Stats" page with the IP address that myMPD use?

This seems a duplicate of the closed issue #107

**Comment by Emilot** [comment]
Sorry i didn't saw it when i made the comment here.

Although it was to nice that i remind it to you and you make it possible so quick!

Awesome! Thank you JC.

**Comment by ghost** [comment]
If Dutch translations are still open, I'm happy to contribute.

**Comment by jcorporation** [comment]
> If Dutch translations are still open, I'm happy to contribute.

This would be great.

**Comment by onkobu** [comment]
I've moved to 6.2.3, with a few hiccups like /etc/mympd/mympd.conf instead of only /etc/mympd.conf and the change in building, but this is the most valuable piece of music interface I've ever made up with. Sadly smart playlists rendered my fancy command line tools useless, putting files by genre into separate lists. Now all I need to do is append Industrial to Nu Jazz and Christian Rap as well as semi-official Winamp extension list code 109 to Hip Hop…

**Comment by jcorporation** [comment]
@onkobu:
Many thanks for your feedback.

> with a few hiccups like /etc/mympd/mympd.conf instead of only /etc/mympd.conf

Should be /etc/mympd.conf in 6.2.3.

**Comment by timpani2012** [comment]
Thanks to jcorporation for building mympd, I contribute Chinese translations.

**Comment by ghost** [comment]
Would it be an option to have just 1 file with all the language strings, instead of 1 file per language? 
If I look at your commit: [https://github.com/jcorporation/myMPD/commit/ce3144582852ac5ab4122c79a75f7e7aefc29bd7]
I see that a string is removed in all language files, but a new 1 is only added in the de-DE file. Translators now always have to compare two files.

If you use 1 file, it is easier to see the missing translations and easier to maintain:
Example:

en-US  Rescan database
de-DE  Datenbank neu einlesen
nl-NL   Herscan database
ko-KR  데이터베이스 다시 검색

en-US  Update directory
de-DE  Verzeichnis aktualisieren
nl-NL   Update map
ko-KR  디렉터리 업데이트

en-US  Error in response to command: %{command}
de-DE  Fehler beim Ausführen von %{command}

**Comment by timpani2012** [comment]
I try to modify it.thanks.----- 原始邮件 -----
发件人：Martins <notifications@github.com>
收件人：jcorporation/myMPD <myMPD@noreply.github.com>
抄送人：timpani2012 <xiaoqingyong@sina.com>,  Comment <comment@noreply.github.com>
主题：Re: [jcorporation/myMPD] Help to improve myMPD (#167)
日期：2020年04月09日 03点47分



Would it be an option to have just 1 file with all the language strings, instead of 1 file per language?


If I look at your commit: [https://github.com/jcorporation/myMPD/commit/ce3144582852ac5ab4122c79a75f7e7aefc29bd7]


I see that a string is removed in all language files, but a new 1 is only added in the de-DE file. Translators now always have to compare two files.

If you use 1 file, it is easier to see the missing translations and easier to maintain:


Example:

en-US  Rescan database


de-DE  Datenbank neu einlesen


nl-NL   Herscan database


ko-KR  데이터베이스 다시 검색

en-US  Update directory


de-DE  Verzeichnis aktualisieren


nl-NL   Update map


ko-KR  디렉터리 업데이트

en-US  Error in response to command: %{command}


de-DE  Fehler beim Ausführen von %{command}

**Comment by jcorporation** [comment]
There is a simple way to show which translations are missing or obsolete.

In devel you can now use `./build.sh translate` to build the translation file for debug builds. This command also shows all missing translations. I hope this makes maintaining translations simpler.
I also added this option to the wiki: https://github.com/jcorporation/myMPD/wiki/Translating

**Comment by ghost** [comment]
> se `./build.sh translate`
Great, even a better solution!

**Comment by Daspunk** [comment]
First, thanks for the work you've done.

I have a rough Finnish translation file done, and looking at some improvements.

Working on a project involving your work.

**Comment by jcorporation** [comment]
Translations are always welcome also rough ones. 

I'm curious, what kind of project?

**Comment by Daspunk** [comment]
I'm working on an automated background music player based on mpd, and myMPD
has proved to be the best overall support interface.

I'm supposed to have a closed network loop across the country for all my
music players, so i can troubleshoot with myMPD.

So i'm more than happy to help with everything i can.

We're a small startup in Finland and i'm pretty much the only developer :)

And sorry, deleted the original reply that I made to the email. Been up for 20 hours :)

**Comment by Daspunk** [comment]
I'll have the finished translation next week, after i get feedback from users

**Comment by jdr85** [comment]
Dear Jürgen, 
I've just compiled and installed myMPD on an old RPi 1B with Archlinux ARM. Compilation and installation was very smooth and took only something between 5 and 10 minutes even on this slow device.

The Web GUI is very responsive, I did not expect this! On my computer it looks great and even displays embedded ID3 Album Art. 

I just have issues on the smartphone and tablet. It does not scale too well. I have to scroll all the time, both in portrait and in landscape mode. The only workaround yet was to change the albumart size in settings, it would be great if scaling worked in a way that the whole page is displayed without scrolling. Would this be possible? There are so many empty spaces and large borders.

Screen resolution here is 720x1080.

Have yet to test everything else.

**Comment by jcorporation** [comment]
@jdr85: thanks for your feedback. If you want, you can open a new issue to discuss this scaling problem. I don't want lengthy discussions in this thread.

Edit:
Commit fb730b2f (v6.5.0 branch) adds a scale ratio setting for mobile browsers. This setting is saved in the local storage of the browser and changes the viewport metatag on the fly.

**Comment by jdr85** [comment]
> 
> 
> @jdr85: thanks for your feedback. If you want, you can open a new issue to discuss this scaling problem. I don't want lengthy discussions in this thread.
> 
> Edit:
> Commit [fb730b2](https://github.com/jcorporation/myMPD/commit/fb730b2f23afd6ea7675a392ff600c02877c55cb) (v6.5.0 branch) adds a scale ratio setting for mobile browsers. This setting is saved in the local storage of the browser and changes the viewport metatag on the fly.

Cool thanks, I have removed my large screenshots as they kind of destroyed the layout of this generic thread.

**Comment by john-** [comment]
Great job on UI changes with 8.6.   I find it a lot more straightforward to browse and search my relatively extensive library.

One minor thing about the "play" icon.   It appears in 3 different places and all have different meaning:

Upper left corner:  Settings
Left side: Show what is currently playing
Lower left: Start playing

After a few minutes of use I figured out what each does.   However, those using mympd more casually may get confused.

A suggestion would be to have settings icon be a hamburger icon or something else that denotes dropdown menu.   The current icon looks cool but with the bold looking green and circle around it it looks like it is the thing that should start making music.

The Playback icon could be something that gives more impression of "now playing".    I did a quick search for ideas and one idea is something that like VU meter (spectrum graph).

**Comment by jcorporation** [comment]
> Great job on UI changes with 8.6

@john-: Thanks for your feedback, you are in the future!

Upper left corner play icon:

Settings - this is the current logo of myMPD. I always wanted to design a fresh logo, but I had no great idea how it should look like. Any ideas are welcome! The icon is in the highlight color, if myMPD is connected.

Left side play icon:

Show what is currently playing - you can change this icon yourself by editing `/var/lib/mympd/state/navbar_icons`. myMPD uses icons from https://material.io/resources/icons/. If you want, you can experiment with other icons, eventually the ligature `equalizer` or `graphic_eq` is better?

**Comment by john-** [comment]
I didn't know that was the mympd logo...for sure that should stay as it is.

I like `equalizer` and `graphic_eq`.  Also, maybe `hearing` would imply "here is what is being listened to".

Like you said this can be customized by the user so if there is no way you get close to what the user wants they can always change it.

**Comment by Emilot** [comment]
Hi JC!

1) I think a sticky top header when someone is scrolling down will be useful in "long" libraries
2) Is there any reason we do not have "Any tag" tag at "Library" card? It is a very user-friendly tag :)

**Reply by jcorporation** [reply]
1) a sticky header can be problematic on small devices

2) this was not possible for the old album query implementation, but possible for the new one.

**Reply by Emilot** [reply]
Great! 

Maybe keep it disable to small (mobile) screens and available for desktop screens(resolutions)?

**Reply by jcorporation** [reply]
This could be an option.

**Comment by jcorporation** [comment]
Please open a new discussion as "Idea" for each proposed enhancement. This discussion is only for general feedback.

**Reply by Emilot** [reply]
Ok JC! Thank you.

**Comment by Emilot** [comment]
Hi JC,

What are the limitations for the search function?

Right now i cannot find a song if i remember only a part of the title. 

ex. I want to find  "While My Guitar Gently Weeps" and i remember "guitar" and "weeps". I cannot find it. No results

I can find it only if i search "guitar gently" or " gently weeps"...etc.

Is there anything we can do to make it more user friendly?

Am i missing something?

**Reply by jcorporation** [reply]
The search function uses the mpd search expressions and has exactly this limitations. You can use a regex for your search: `guitar .*weeps` or you can search for `guiter` and `weeps.`

![image](https://user-images.githubusercontent.com/765844/110248661-8cca4e80-7f72-11eb-98dd-8c2256cdd12a.png)

**Reply by Emilot** [reply]
Thank you JC.

with "guitar .*weeps" nothing found. When you say to search for "guitar" AND "weeps" what do you mean? Can i include both in one search expression?

**Reply by jcorporation** [reply]
You must select `=~`as comparison operator for `guitar .*weeps`to work.

For the second option, look at the screenshot: 
1. Search for "guitar" -> press enter
2. Search for "weeps"

Please open a extra discussion in the future. This thread is for general "help to improve" topics.

**Reply by Emilot** [reply]
Ok JC. Thank you!

**Comment by dimafon52** [comment]
Thank You for your work!
I found a problem. Please tell me how to solve it.
"Feedback" keeps disappearing
![mympd_feedback](https://github.com/user-attachments/assets/8f03af4e-39c1-400b-a174-3652244fd1a3)
 
After: systemctl restart mympd - it's OK. 

Thank you!

**Reply by jcorporation** [reply]
This is not the right place to report a problem. If you think this is a bug zehn open an issue, else ask in the Q&A section for help.

For both scenarios: A detailed failure description is important. Debug logs, myMPD versionen and config, MPD config, etc.
